src/main/java/mate.academy/service/impl/ProductServiceImpl.java

```diff
@@ -2,13 +2,17 @@
 
 import java.util.List;
 import java.util.stream.Collectors;
+import mate.academy.lib.Component;
 import mate.academy.model.Product;
 import mate.academy.service.FileReaderService;
 import mate.academy.service.ProductParser;
 import mate.academy.service.ProductService;
 
+@Component
```

Comment:
if you didn't annotate your other implementations with @Component - you shouldn't be able initializing using them in injector - it has to be exception